client.go

```diff
@@ -321,7 +326,11 @@ func (nclient *NativeClient) Connect() (*ssh.Client, *SessionInfo, error) {
 		destAddr = fmt.Sprintf("%s:%d", h.HostName, h.Port)
 		if sshClient == nil {
 			//first host
+			// save the original timeout if we are passing a non-default one
+			defTimout = h.ClientConfig.Timeout
+			h.ClientConfig.Timeout = timeout
```

Comment:
It's probably unlikely.. but this does introduce a potential race condition in which some other thread which is using this client will get its timeout changed if it happens to run while the Dial is happening.    You might consider adding a "CopyWithTimeout" function as an alternative.  There's already a Copy that does most of what is needed, you'd just need to override the timeout.

Reply:
You are right - I think it's safer to make a copy of the client and use that instead. Let me do that.